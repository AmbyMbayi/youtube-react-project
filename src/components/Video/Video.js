import './Video';

import React from 'react';

const BASE_EMBED_URL = 'https://www.youtube.com/embed/';
function Video(props) {
	if (!props.id) {
		return null;
	}
	const embedurl = `${BASE_EMBED_URL}${props.id}`;
	//const embedurl = `${BASE_EMBED_URL}${props.id}?autoplay=1`;
	return (
		<iframe
			src={embedurl}
			width={'560'}
			height={'315'}
			frameBorder="0"
			allow="autoplay; encrypted-media"
			allowFullScreen
			title={'video'}
		/>
	);
}

export default Video;
