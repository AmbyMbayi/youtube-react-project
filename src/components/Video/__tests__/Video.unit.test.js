import React from 'react';
import Video from '../Video';
import { shallow } from 'enzyme';

test('renders video component correctly', () => {
	const wrapper = shallow(<Video id="HAuXJVI_bUs" />);
	expect(wrapper).toMatchSnapshot();
});

test('renders nullif id in video componenent is not specified', () => {
	const wrapper = shallow(<Video />);
	expect(wrapper).toMatchSnapshot();
});
