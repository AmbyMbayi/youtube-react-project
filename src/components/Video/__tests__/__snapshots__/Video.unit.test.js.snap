// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nullif id in video componenent is not specified 1`] = `""`;

exports[`renders video component correctly 1`] = `
<iframe
  allow="autoplay; encrypted-media"
  allowFullScreen={true}
  frameBorder="0"
  height="315"
  src="https://www.youtube.com/embed/HAuXJVI_bUs"
  title="video"
  width="560"
/>
`;
